src/platform/plugins/shared/discover/public/application/main/state_management/redux/selectors/tabs.ts

```diff
@@ -9,11 +9,10 @@
 
 import { createSelector } from '@reduxjs/toolkit';
 import type { DiscoverInternalState } from '../types';
-import { TabsBarVisibility, type TabState } from '../types';
+import { TabsBarVisibility } from '../types';
 
-export const selectTab = (state: DiscoverInternalState, tabId: string) => state.tabs.byId[tabId];
-export const selectOpenTab = (state: DiscoverInternalState, tabId: string): TabState | undefined =>
-  state.tabs.allIds.includes(tabId) ? state.tabs.byId[tabId] : undefined;
+export const selectTab = (state: DiscoverInternalState, tabId: string) =>
+  state.tabs.byId[tabId] ?? state.tabs.recentlyClosedTabsById[tabId];
```

Comment:
Now we can resolve the `selectTab` from open > closed case directly in the selector...